### ISSUE TITLE
Pointer Lock Unadjusted Movement option

### ISSUE THREAD

**Issue by aprotyas** [WebKit/standards-positions#254]
### WebKittens

@marcoscaceres

### Title of the spec

Pointer Lock Unadjusted Movement option

### URL to the spec

https://github.com/w3c/pointerlock/pull/49

### URL to the spec's repository

https://github.com/w3c/pointerlock

### Mozilla standards-positions issue URL

https://github.com/mozilla/standards-positions/issues/448

### WebKit Bugzilla URL

https://bugs.webkit.org/show_bug.cgi?id=218054

### Description

This is an unmerged proposal that was discussed in this issue:  https://github.com/w3c/pointerlock/issues/36

This feature is requested by gaming platforms on the web, and there is a work-in-progress implementation submitted by James Howard here: https://github.com/WebKit/WebKit/pull/17525.

Among other things, this proposed change introduces a Promise workflow.

**Comment by aprotyas** [comment]
James Hollyer had requested a position on the webkit-dev mailing list: https://lists.webkit.org/pipermail/webkit-dev/2020-October/031473.html, but let's migrate the discourse here now.

**Comment by annevk** [comment]
@NavidZ @james-howard what is the status of https://github.com/w3c/pointerlock/pull/49? It seems like it hasn't been updated for over a year. This feature on its own seems reasonable, but it would be good if there was a finished specification.

**Comment by NavidZ** [comment]
@mustaqahmed might know better of the status. To the best of my knowledge, it was already implemented at least in Chromium.

**Comment by james-howard** [comment]
@annevk here's the situation as I understand it:

@marcoscaceres [Said](https://github.com/WebKit/WebKit/pull/17525#discussion_r1319199834) that WPT tests are a pre-req for merging the specification. I added some in my [WebKit PR](https://github.com/WebKit/WebKit/pull/17525/files#diff-b58cf136571fbc36897e376a80407cf2d2d48c09e90b5465609cb074a2c8b198) with the plan that they could be upstreamed to the WPT.

@reillyeon [Said](https://github.com/w3c/pointerlock/pull/49#issuecomment-1041005606) that having an implementation outside of Chromium would help. I've written a WebKit implementation that is under review. Please let's try to avoid deadlock here (can't merge the spec without multiple implementations, can't get multiple implementations without merging the spec).

**Comment by annevk** [comment]
The spec PR was last touched a year-and-half ago, has merge conflicts, and has multiple requests for changes. Those should be resolved in some way and none of that is blocked by a second implementation as far as I can tell.

**Comment by james-howard** [comment]
To my reading, the requests for changes in the spec are just about formatting and style. I don't see any open requests that would affect the implementation in my WebKit PR.

Can you clarify the order you would like to see things done in? Is getting the spec cleaned up and merged a prerequisite to getting my WebKit PR accepted? Is it a prerequisite to getting a standards position staked out here?

**Comment by annevk** [comment]
Landing something in WebKit behind a preference is not blocked on standards-positions. Enabling it by default can be though.

**Comment by mustaqahmed** [comment]
> The spec PR was last touched a year-and-half ago, has merge conflicts, and has multiple requests for changes. Those should be resolved in some way and none of that is blocked by a second implementation as far as I can tell.

I agree.  Let's continue the PR discussion in https://github.com/w3c/pointerlock/pull/49.

> To the best of my knowledge, it was already implemented at least in Chromium.

That's correct.

**Comment by naderslr** [comment]
Any update on this? I have my game disabled on Firefox but I have a lot of users that want Firefox support. Any update on when is this ever planned on the future?

I'm looking for the unadjustmentMovement on Firefox pointerlock.
